src/main/java/org/jenkinsci/plugins/workflow/steps/CoreWrapperStep.java

```diff
@@ -92,20 +94,82 @@ public SimpleBuildWrapper getDelegate() {
+            }
+            final StepContext context = getContext();
+            final Run<?, ?> run = context.get(Run.class);
+            assert run != null;
+            {
+                final TaskListener listener = context.get(TaskListener.class);
+                assert listener != null;
+                final EnvVars env = context.get(EnvVars.class);
+                assert env != null;
+                final FilePath workspace = context.get(FilePath.class);
+                final Launcher launcher = context.get(Launcher.class);
+                boolean workspaceRequired = true;
+                // In Jenkins 2.25x, a SimpleBuildWrapper can indicate that it does not require a workspace context by
+                // overriding a requiresWorkspace() method to return false. So use that if it's available.
+                // Note: this uses getMethod() on the delegate's type and not SimpleBuildWrapper so that an
+                // implementation can get this behaviour without switching to Jenkins 2.25x itself.
+                // TODO: Use 'workspaceRequired = this.delegate.requiresWorkspace()' once this plugin depends on Jenkins 2.25x or later.
+                try {
+                    final Method requiresWorkspace = this.delegate.getClass().getMethod("requiresWorkspace");
+                    workspaceRequired = (boolean) requiresWorkspace.invoke(this.delegate);
+                } catch(NoSuchMethodException e) {
+                    // ok, default to true
+                } catch (InvocationTargetException ite) {
+                    final Throwable realException = ite.getCause();
+                    throw realException instanceof Exception ? (Exception) realException : ite;
+                }
+                if (workspaceRequired) {
+                    if (workspace == null) {
+                        throw new MissingContextVariableException(FilePath.class);
+                    }
+                    if (launcher == null) {
+                        throw new MissingContextVariableException(Launcher.class);
+                    }
+                }
+                // always pass the workspace context when available, even when it is not strictly required
+                if (workspace != null && launcher != null) {
+                    this.delegate.setUp(c, run, workspace, launcher, listener, env);
+                } else {
+                    // If we get here, workspaceRequired is false and there is no workspace context. In that case, the
+                    // overload of setUp() introduced in Jenkins 2.25x MUST exist.
+                    // Note: this uses getMethod() on the delegate's type and not SimpleBuildWrapper so that an
+                    // implementation can get this behaviour without switching to Jenkins 2.25x itself.
+                    // TODO: Use 'this.delegate.setUp(c, run, listener, env)' once the minimum core version for this plugin is 2.25x or newer.
```

Comment:
```suggestion
                    // overload of setUp() introduced in Jenkins 2.258 MUST exist.
                    // Note: this uses getMethod() on the delegate's type and not SimpleBuildWrapper so that an
                    // implementation can get this behaviour without switching to Jenkins 2.258 itself.
                    // TODO: Use 'this.delegate.setUp(c, run, listener, env)' once the minimum core version for this plugin is 2.258 or newer.
```